### PR TITLE
enable Non-STS and do structure adjustment

### PR DESCRIPTION
**Pull request by yasun1**
close [OCPQE-15386](https://issues.redhat.com/browse/OCPQE-15386)

### CODE REVIEW

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-rosa-sts-hypershift-guest-integration-full-p1-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-aws-rosa-sts-p2-f7

**Comment by yasun1** [comment]
@DennisPeriquet To easily manage the workflow, we organized the provision/deprovision steps into chains. Please help review with the [aws-load-balancer-operator-release](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/41799/rehearse-41799-pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator/1685954681123115008) log.

File: ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh [hunk 1]
```diff
@@ -43,10 +43,13 @@ while rosa describe cluster -c "${CLUSTER_ID}" ; do
   sleep 60
 done
 
-echo "Deleting operator roles"
-rosa delete operator-roles -c "${CLUSTER_ID}" -y -m auto
-echo "Deleting oidc-provider"
-rosa delete oidc-provider -c "${CLUSTER_ID}" -y -m auto
+if [[ "$STS" == "true" ]]; then
```
Comment (yuwang-RH) [inline ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh:46]: will this script use for hosted-cp cluster deprovision in future?If yes, operator-roles and oidc-provider need to clean for hosted-cp clusters too. Or we treat hosted-cp cluster as STS cluster too?

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh [hunk 2]
```diff
@@ -129,7 +135,6 @@ if [[ "$DISABLE_WORKLOAD_MONITORING" == "true" ]]; then
 fi
 
 HYPERSHIFT_SWITCH=""
-SUBNET_ID_SWITCH=""
 if [[ "$HOSTED_CP" == "true" ]]; then
   HYPERSHIFT_SWITCH="--hosted-cp --classic-oidc-config"
```
Comment (yuwang-RH) [inline ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:139]: classic-oidc-config is not the suggested option, it is hidden. It still can be used, it depends on you if use the the byo oidc config

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh [hunk 3]
```diff
@@ -223,33 +214,42 @@ if [[ "$ENABLE_BYOVPC" == "true" ]]; then
...
+  Account_Support_Role_Name="${ACCOUNT_ROLES_PREFIX}-Support-Role"
+  Account_Worker_Role_Name="${ACCOUNT_ROLES_PREFIX}-Worker-Role"
+
+  roleARNList=$(rosa list account-roles -o json | jq -r '.[].RoleARN')
+
+  Account_Installer_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Installer_Role_Name}" || true; })
+  Account_ControlPlane_Role_ARN=$(echo "$roleARNList" | { grep "${Account_ControlPlane_Role_Name}" || true; })
+  Account_Support_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Support_Role_Name}" || true; })
+  Account_Worker_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Worker_Role_Name}" || true; })
+  if [[ -z "${Account_ControlPlane_Role_ARN}" ]] || [[ -z "${Account_Installer_Role_ARN}" ]] || [[ -z "${Account_Support_Role_ARN}" ]] || [[ -z "${Account_Worker_Role_ARN}" ]]; then
```
Comment (yuwang-RH) [inline ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:237]: There is no 'ControlPlane' for hosted-cp cluster. The related flag is still can be set. One potential problem is that no ControlPlane role created if use the `rosa create account-roles --hosted-cp` command.

**Review by yuwang-RH — COMMENTED** [review]
/lgtm

File: ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh (hunk 1 above)
Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh:46]: The deprovision is used for any kind of cluster.

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh (hunk 2 above)
Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:139]: It is just a workaround. We need to do more around the OIDC config.

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh (hunk 3 above)
Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:237]: First time to know that, let us talk about this later.

**Comment by yasun1** [comment]
@alebedev87 @Miciah @candita @miheer @rfredette @gcs278 @frobware @thejasn Hi guys, could one of you review the change on the workflow `aws-load-balancer-install-rosa`

> To easily manage the workflow, we organized the provision/deprovision steps into chains. Please help review with the [aws-load-balancer-operator-release](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/41799/rehearse-41799-pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator/1685954681123115008) log.

**Comment by heliubj18** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2

**Comment by heliubj18** [comment]
@yasun1 job rehearsal failed
```
INFO[2023-08-02T03:29:46Z] Running step aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2-rosa-sts-account-roles-create. 
INFO[2023-08-02T03:29:57Z] Logs for container test in pod aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2-rosa-sts-account-roles-create: 
INFO[2023-08-02T03:29:57Z] Logging into production with offline token
rosa: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by rosa)
rosa: /lib64/libc.so.6: version `GLIBC_2.[34](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41799/rehearse-41799-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2/1686578587198558208#1:build-log.txt%3A34)' not found (required by rosa)
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:84","func":"k8s.io/test-infra/prow/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2023-08-02T03:29:57Z"}
error: failed to execute wrapped command: exit status 1 
INFO[2023-08-02T03:29:57Z] Step aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2-rosa-sts-account-roles-create failed after 10s.
```

**Comment by heliubj18** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator

**Comment by heliubj18** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-rosa-sts-hypershift-sector-guest-prod-critical-p1-f2

**Comment by heliubj18** [comment]
LGTM

**Comment by heliubj18** [comment]
The test for hosted cluster e2e passed. Thanks @yasun1

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-aws-load-balancer-operator-release-1.0-e2e-aws-rosa-operator

**Pull request merged**